pyironflow/wf_extensions.py

```diff
@@ -35,6 +35,11 @@ def dict_to_node(dict_node):
         target_labels = data['target_labels']
         for k, v in zip(target_labels, target_values):
             if v not in ('NonPrimitive', 'NotData'):
+                type_hint = node.inputs[k].type_hint
+                # JS gui can return input values like 2.0 as int, breaking type hints
+                # so check here if the type hint is a float, but convert only if losslessly possible
+                if not valid_value(v, type_hint) and valid_value(float(v), type_hint) and v == float(v):
```

Comment:
I know with comparing two floats the strict equality is dangerous. For int to float comparison I'm not sure.

Reply:
I should probably make this clearer in the code, but the issue is only triggered if `type(v) == int` and in that case I thought python guarantees the equality.  Anyway, I check all the numbers until `1080454053`, for fun.  Wikipedia tells me as long as the integer is not wider than the mantissa of the IEEE floats it's converted to, it'll be fine.  I'm declaring that 52bit integers are enough should be enough for the GUI. ;)